msbuild/Xamarin.MacDev.Tasks/Tasks/AOTCompileTaskBase.cs

```diff
@@ -97,11 +100,22 @@ public override bool Execute ()
 					Log.LogError (MSBStrings.E7071, /* Unable to parse the AOT compiler arguments: {0} ({1}) */ processArguments, ex2!.Message);
 					return false;
 				}
-				arguments.Add ($"{string.Join (",", parsedArguments)}");
-				if (globalAotArguments is not null)
+				string dedupArgument = "";
+				if (Path.GetFileName (input) == Path.GetFileName (DedupAssembly)) {
+					dedupArgument = $",dedup-include={Path.GetFileName (DedupAssembly)}";
+				} else if (!string.IsNullOrEmpty (DedupAssembly)) {
+					dedupArgument = ",dedup-skip";
+				}
```

Comment:
Finally, the code around here would be moved to `ComputeAOTArguments` linker step.

Reply:
Afaik `ComputeAOTArguments` is a configuration aware step which has access to: https://github.com/xamarin/xamarin-macios/blob/8b3a4340001103690b9f954dea883fc96e6b3909/tools/dotnet-linker/LinkerConfiguration.cs#L18 
This configuration object gets populated by a build-time generated file: `custom-linker-options.txt` generated in the `_ComputeLinkerArguments` MSBuild target, so I suppose you could:
- introduce another property on the LinkerConfiguration type, or 
- add item metadata to the `_DedupAssembly` which you can access in the custom linker step